### PR TITLE
implemented Bank services subscribers report

### LINKED ISSUE #71: Bank services subscribers report
1. Table bank_services_subscriber: id (long), one_time_amount, weekly_amount, monthly_amount (int)
2. Entity, repository...
3. Create a service with a job that runs every day and aggregates amount of subscribers for every type of bank service.
4. Create an endpoint GET /bank/services/subscribers that returns the newest record in the bank_services_subscribers table.

### PR DESCRIPTION
**Pull request by andy-sam**
fixes #71

### CODE REVIEW

**Review by alex-anakin — CHANGES_REQUESTED** [review]

File: src/main/java/com/codesarge/banking/repositories/BankServicesSubscribersRepository.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.codesarge.banking.entities.BankServicesSubscriber;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import java.time.LocalDate;
+import java.util.List;
+
+public interface BankServicesSubscribersRepository extends JpaRepository<BankServicesSubscriber, Long> {
+
+    @Query(value = " SELECT  counting_date AS ReportingDate, one_time_amount AS OneTimeAmount,weekly_amount AS WeeklyAmount,monthly_amount AS MonthlyAmount FROM bank_services_subscriber " +
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/repositories/BankServicesSubscribersRepository.java:11]: done

File: src/main/java/com/codesarge/banking/repositories/BankServicesSubscribersRepository.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            "ORDER BY counting_date DESC LIMIT 1;", nativeQuery = true)
+    List<BankServicesSubscriberResult> countTypesOfBankServicesSubscribers();
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/repositories/BankServicesSubscribersRepository.java:13]: done

File: src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                    nullable: false
+              - column:
+                  name: one_time_amount
+                  type: BIGINT
+              - column:
+                  name: monthly_amount
+                  type: BIGINT
+              - column:
+                  name: weekly_amount
+                  type: BIGINT
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml:29]: done

File: src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: one_time_amount
+                  type: BIGINT
+              - column:
+                  name: monthly_amount
+                  type: BIGINT
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml:26]: done

File: src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+              - column:
+                  name: id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: one_time_amount
+                  type: BIGINT
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/resources/db/changelog/changelog-2021-09-10-bankServiceSubscriber.yaml:23]: done

File: src/main/java/com/codesarge/banking/repositories/SubscriptionRepository.java [hunk 6]
```diff
@@ -14,4 +15,14 @@
...
+
+    @Query(value = "select count(bank_service_id) from subscription where bank_service_id = 1",
+            nativeQuery = true)
+    int countOfOneTimeSubscriptions();
+    @Query(value = "select count(bank_service_id) from subscription where bank_service_id = 2",
+            nativeQuery = true)
+    int countOfWeeklySubscriptions();
+    @Query(value = "select count(bank_service_id) from subscription where bank_service_id = 3",
+            nativeQuery = true)
+    int countOfMonthlySubscriptions();
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/repositories/SubscriptionRepository.java:27]: done

**Review by alex-anakin — APPROVED** [review]

**Pull request merged**